eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -137,9 +137,31 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     }
 }
 
+$verifyDeleteScript = {
+    try {
+        $group = Get-AzResourceGroup -name $ResourceGroupName
+    } catch {
+        if ($Error[0].ToString().Contains("Provided resource group does not exist")) {
+            Write-Verbose "Resource group '$ResourceGroupName' not found. Continuing..."
+            return
+        }
+        throw $Error[0]
+    }
+
+    if ($group.ProvisioningState -ne "Deleting")
+    {
+        throw "Resource group is in '$($group.ProvisioningState)' state, expected 'Deleting'"
```

Comment:
Do we expect that this will fail on the first try everytime? Or does it generally return deleting on the first try?

Reply:
Generally it returns deleting, but in the case it doesn't, this triggers the first loop in `Retry()` which sleeps for 10 seconds then tries again.

Reply:
The ARM resource group operations are pretty fast in my experience.